lib/net.js

```diff
@@ -899,24 +899,18 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-Socket.prototype.connect = function(options, cb) {
+Socket.prototype.connect = function() {
+  const args = new Array(arguments.length);
+  for (var i = 0; i < arguments.length; i++)
+    args[i] = arguments[i];
+  // TODO(joyeecheung): use destructuring when V8 is fast enough
```

Comment:
Yup. but anyway we need to `normalizeArgs` on `exports.createConnection` since arguments of `new Socket` needs normalized arguments.

I will just add one more commit to move timeout from `exports.createConnection` to `Socket.prototype.connect`.

After that, maybe it should be good to merge?